File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/LanguageEnum.java
package ai.ragu.charactermanager.enumeration;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.HashMap;
import java.util.Map;

public enum LanguageEnum {
    AQUAN("aquan"),
    BOGGARD("boggard"),
    COMMON("common"),
    CELESTIAL("celestial"),
    DWARVEN("dwarven"),
    DRACONIC("draconic"),
    ELVEN("elven"),
    GNOMISH("gnomish"),
    GNOLL("gnoll"),
    GOBLIN("goblin"),
    HALFLING("halfling"),
    IRUXI("iruxi"),
    JOTUN("jotun"),
    ORCISH("orcish"),
    SYLVAN("sylvan"),
    TERRAN("terran"),
    UNDERCOMMON("undercommon");

    private final String value;
    private final static Map<String, LanguageEnum> CONSTANTS = new HashMap<>();

    static {
        for (LanguageEnum c: values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    LanguageEnum(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }

    @JsonCreator
    public static LanguageEnum fromValue(String value) {
        LanguageEnum constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new IllegalArgumentException(value);
        } else {
            return constant;
        }
    }
}

File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/LanguageEnum.kt
package ai.ragu.charactermanager.enumeration

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import java.util.*

enum class LanguageEnum(private val value: String) {
    AQUAN("aquan"), BOGGARD("boggard"), COMMON("common"), CELESTIAL("celestial"), DWARVEN("dwarven"), DRACONIC("draconic"), ELVEN("elven"), GNOMISH("gnomish"), GNOLL("gnoll"), GOBLIN("goblin"), HALFLING("halfling"), IRUXI("iruxi"), JOTUN("jotun"), ORCISH("orcish"), SYLVAN("sylvan"), TERRAN("terran"), UNDERCOMMON("undercommon");

    companion object {
        private val CONSTANTS: MutableMap<String, LanguageEnum> = HashMap()
        @JsonCreator
        fun fromValue(value: String?): LanguageEnum {
            val constant = CONSTANTS[value]
            return constant ?: throw IllegalArgumentException(value)
        }

        init {
            for (c in values()) {
                CONSTANTS[ai.ragu.charactermanager.enumeration.c.value] = ai.ragu.charactermanager.enumeration.c
            }
        }
    }

    override fun toString(): String {
        return value
    }

    @JsonValue
    fun value(): String {
        return value
    }

}